metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/delete_test.clj

```diff
@@ -94,75 +81,112 @@
           gran1 (util/create-and-save-granule provider-id coll1 1 2)
           coll2 (util/create-and-save-collection provider-id 2)
           gran3 (util/create-and-save-granule provider-id coll2 2)
-          {:keys [status revision-id] :as tombstone} (util/delete-concept (:concept-id coll1))
-          deleted-coll1 (:concept (util/get-concept-by-id-and-revision (:concept-id coll1) revision-id))
-          saved-coll1 (:concept (util/get-concept-by-id-and-revision (:concept-id coll1) (dec revision-id)))]
-      (is (= {:status 201
-              :revision-id 4}
-             {:status status
-              :revision-id revision-id}))
-
-      (is (= (dissoc (assoc saved-coll1
-                            :deleted true
-                            :metadata ""
-                            :revision-id revision-id
-                            :user-id nil)
-                     :revision-date :transaction-id)
-             (dissoc deleted-coll1 :revision-date :transaction-id)))
-
-      ;; Make sure that a deleted collection gets it's own unique revision date
-      (is (t/after? (:revision-date deleted-coll1) (:revision-date saved-coll1))
-          "The deleted collection revision date should be after the previous revisions revision date.")
-
-      ;; Verify granule was deleted
-      (is (= {:status 404} (util/get-concept-by-id-and-revision (:concept-id gran1) 1)))
-      (is (= {:status 404} (util/get-concept-by-id-and-revision (:concept-id gran1) 2)))
-
-      ;; Other data left in database
-      (util/verify-concept-was-saved coll2)
-      (util/verify-concept-was-saved gran3))))
+          variable (util/create-and-save-variable "REG_PROV" 1)
+          variable-association (util/create-and-save-variable-association coll1 variable 1 1)
+          service (util/create-and-save-service provider-id 1)
+          service-association (util/create-and-save-service-association coll1 service 1 1)]
+      
+      ;; variable association is not a tombstone before collection is deleted
+      (is (= false (is-association-tombstone? variable-association)))
+      ;; service association is not a tombstone before collection is deleted
+      (is (= false (is-association-tombstone? service-association)))
+
+      ;; now delete coll1
+      (let [{:keys [status revision-id]} (util/delete-concept (:concept-id coll1))
+            deleted-coll1 (:concept (util/get-concept-by-id-and-revision
+                                     (:concept-id coll1) revision-id))
+            saved-coll1 (:concept (util/get-concept-by-id-and-revision
+                                   (:concept-id coll1) (dec revision-id)))]
+
+        (is (= {:status 201
+                :revision-id 4}
+               {:status status
+                :revision-id revision-id}))
+
+        (is (= (dissoc (assoc saved-coll1
```

Comment:
It might be easier to read by putting this logic into an assert-collection-tombstone function, especially since the same logic is repeated further down in this test file.

Reply:
Done.